### PR TITLE
Fix some violations to boolean variable naming convention #7131

### LINKED ISSUE #7131: Fix some violations to boolean variable naming convention
Fix some violations reported by https://xpdavid.github.io/CS2103R-Report/#boolean-variable-naming-convention

### PR DESCRIPTION
**Pull request by xpdavid**
Fixes #7131

**Outline of Solution**
Fix violations based on report
[Main](http://htmlpreview.github.io/?https://github.com/xpdavid/CS2103R-Report/blob/master/codingStandard/booleanNaming/main.html)
[Test](http://htmlpreview.github.io/?https://github.com/xpdavid/CS2103R-Report/blob/master/codingStandard/booleanNaming/test.html)

[Discussion](https://xpdavid.github.io/CS2103R-Report/#boolean-variable-naming-convention) on boolean variable naming convention.

The fixing on `datatransfer` and `entity` would require more efforts as they are related to how data is stored in GAE, typically JSON format. I suspect the side effects may be outweigh their benefits (need a lot data migration etc.).

### CODE REVIEW

File: src/main/java/teammates/ui/controller/Action.java [hunk 1]
```diff
@@ -224,12 +224,12 @@ private boolean isNotLegacyLink() {
     }
 
     private boolean doesUserNeedToLogin(UserType currentUser) {
-        boolean userNeedsGoogleAccountForPage =
+        boolean shouldUserNeedGoogleAccountForPage =
                 !Const.SystemParams.PAGES_ACCESSIBLE_WITHOUT_GOOGLE_LOGIN.contains(request.getRequestURI());
-        boolean userIsNotLoggedIn = currentUser == null;
-        boolean noRegkeyGiven = getRegkeyFromRequest() == null;
+        boolean isUserNotLoggedIn = currentUser == null;
```
Comment (xpdavid) [inline src/main/java/teammates/ui/controller/Action.java]: I though sometimes it is not worth to enforce the negation rule. In this case, only one branch in `if` statement is used. I though `if (isUserNotLoggedIn) {...}` is better than `if (!isUserLoggedIn) { ... }`

Comment (ablyx) [inline src/main/java/teammates/ui/controller/Action.java]: hmm.. alright then. There are also cases of `if (!isUserLoggedIn) { ... }` though

**Comment by whipermr5** [comment]
@ablyx Reminder :)

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 2]
```diff
@@ -548,8 +548,8 @@ public String getValidityInfoForAllowedName(String fieldName, int maxLength, Str
                                             fieldName, REASON_TOO_LONG, maxLength);
         }
         if (!Character.isLetterOrDigit(value.codePointAt(0))) {
-            boolean isStartWithBraces = value.charAt(0) == '{' && value.contains("}");
-            if (!isStartWithBraces) {
+            boolean isBegunWithBraces = value.charAt(0) == '{' && value.contains("}");
```
Comment (ablyx) [inline src/main/java/teammates/common/util/FieldValidator.java]: This still feels a little weird to it, but I cant think of any replacement for it.

Comment (whipermr5) [inline src/main/java/teammates/common/util/FieldValidator.java]: What about `hasStartingBrace`?

File: src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java [hunk 3]
```diff
@@ -298,71 +298,71 @@ public boolean isResponseCommentVisibleForUser(String userEmail, UserRole role,
...
     }
 
     private boolean isVisibleToUserTeam(StudentAttributes student, Set<String> studentsEmailInTeam,
             FeedbackResponseAttributes response, FeedbackQuestionAttributes relatedQuestion,
-            FeedbackResponseCommentAttributes relatedComment, boolean userIsStudent) {
+            FeedbackResponseCommentAttributes relatedComment, boolean isUserStudent) {
 
-        boolean userIsInResponseRecipientTeamAndRelatedResponseCommentIsVisibleToRecipients =
-                userIsStudent
+        boolean isUserInResponseRecipientTeamAndRelatedResponseCommentIsVisibleToRecipients =
```
Comment (whipermr5) [inline src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java]: One `is` should suffice: `isUserInResponseRecipientTeamAndRelatedResponseCommentVisibleToRecipients`. Similar instances below.

File: src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java [hunk 4]
```diff
@@ -298,71 +298,71 @@ public boolean isResponseCommentVisibleForUser(String userEmail, UserRole role,
...
-                || userIsInResponseRecipientTeamAndRelatedResponseCommentIsVisibleToRecipientsTeamMembers;
+        return isUserInResponseRecipientTeamAndRelatedResponseCommentIsVisibleToRecipients
+                || isUserInResponseGiverTeamAndRelatedResponseCommentIsVisibleToGiversTeamMembers
+                || isUserInResponseRecipientTeamAndRelatedResponseCommentIsVisibleToRecipientsTeamMembers;
     }
 
     private boolean isVisibleToUser(String userEmail, FeedbackResponseAttributes response,
             FeedbackQuestionAttributes relatedQuestion, FeedbackResponseCommentAttributes relatedComment,
-            boolean isVisibleToGiver, boolean userIsInstructor, boolean userIsStudent) {
+            boolean isVisibleToGiver, boolean userIsInstructor, boolean isUserStudent) {
```
Comment (whipermr5) [inline src/main/java/teammates/logic/core/FeedbackResponseCommentsLogic.java]: I think you missed this one: `userIsInstructor`

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java [hunk 5]
```diff
@@ -1926,8 +1926,8 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForQuestionId(Stri
                                                     question, userEmail, UserRole.INSTRUCTOR, section);
                 }
 
-                boolean thisQuestionHasResponses = !responsesForThisQn.isEmpty();
-                if (thisQuestionHasResponses) {
+                boolean isThisQuestionHasResponses = !responsesForThisQn.isEmpty();
```
Comment (whipermr5) [inline src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: I think `hasResponses` should suffice.

File: src/main/java/teammates/ui/controller/Action.java (hunk 1 above)
Comment (whipermr5) [inline src/main/java/teammates/ui/controller/Action.java]: I prefer `!isUserLoggedIn`; it's more readable.

File: src/main/java/teammates/ui/controller/Action.java [hunk 6]
```diff
@@ -224,12 +224,12 @@ private boolean isNotLegacyLink() {
...
+        boolean hasNoRegkey = getRegkeyFromRequest() == null;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/controller/Action.java]: I agree; use `hasRegkey` and `!hasRegkey` later on.

File: src/main/java/teammates/ui/controller/Action.java [hunk 7]
```diff
@@ -224,12 +224,12 @@ private boolean isNotLegacyLink() {
     }
 
     private boolean doesUserNeedToLogin(UserType currentUser) {
-        boolean userNeedsGoogleAccountForPage =
+        boolean shouldUserNeedGoogleAccountForPage =
```
Comment (whipermr5) [inline src/main/java/teammates/ui/controller/Action.java]: `isGoogleLoginRequired` would be better.

File: src/main/java/teammates/ui/controller/InstructorFeedbackResponseCommentsLoadAction.java [hunk 8]
```diff
@@ -78,9 +78,9 @@ private void removeQuestionsAndResponsesIfNotAllowed(FeedbackSessionResultsBundl
                     instructor.isAllowedForPrivilege(fdr.recipientSection, fdr.feedbackSessionName,
                                        Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS);
 
-            boolean instructorHasSessionViewingPrivileges = canInstructorViewSessionInGiverSection
+            boolean shouldInstructorHaveSessionViewingPrivileges = canInstructorViewSessionInGiverSection
```
Comment (whipermr5) [inline src/main/java/teammates/ui/controller/InstructorFeedbackResponseCommentsLoadAction.java]: `hasSessionViewingPrivileges` should suffice.

File: src/main/java/teammates/ui/controller/InstructorFeedbackSubmissionEditPageAction.java [hunk 9]
```diff
@@ -18,8 +18,8 @@ protected boolean isSpecificUserJoinedCourse() {
     @Override
     protected void verifyAccesibleForSpecificUser(FeedbackSessionAttributes session) {
         InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, account.googleId);
-        boolean creatorOnly = false;
-        gateKeeper.verifyAccessible(instructor, session, creatorOnly);
+        boolean isEnabledForCreatorOnly = false;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/controller/InstructorFeedbackSubmissionEditPageAction.java]: Use `isCreatorOnly` for consistency with [InstructorFeedbackResultsPageAction.java](https://github.com/xpdavid/teammates/blob/1bf08cd6ad9ff8e83025289cfc73b7380d3ffc81/src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageAction.java#L38). Also, update the function definition in `GateKeeper.java`.

File: src/main/java/teammates/ui/controller/InstructorFeedbackSubmissionEditSaveAction.java [hunk 10]
```diff
@@ -17,8 +17,8 @@
     protected void verifyAccesibleForSpecificUser() {
         InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, account.googleId);
         FeedbackSessionAttributes session = logic.getFeedbackSession(feedbackSessionName, courseId);
-        boolean creatorOnly = false;
-        gateKeeper.verifyAccessible(instructor, session, creatorOnly);
+        boolean isEnabledForCreatorOnly = false;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/controller/InstructorFeedbackSubmissionEditSaveAction.java]: Same as above.

File: src/main/java/teammates/ui/template/CommentRow.java [hunk 11]
```diff
@@ -13,17 +13,17 @@
 
     private String extraClass = "";
 
-    private boolean withVisibilityIcon;
+    private boolean isWithVisibilityIcon;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/CommentRow.java]: `hasVisibilityIcon`?

File: src/main/java/teammates/ui/template/CommentRow.java [hunk 12]
```diff
@@ -13,17 +13,17 @@
...
     private String whoCanSeeComment;
 
-    private boolean withNotificationIcon;
+    private boolean isWithNotificationIcon;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/CommentRow.java]: `hasNotificationIcon`?

File: src/main/java/teammates/ui/template/CommentRow.java [hunk 13]
```diff
@@ -13,17 +13,17 @@
...
 
-    private boolean withLinkToCommentsPage;
+    private boolean isWithLinkToCommentsPage;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/CommentRow.java]: `hasLinkToCommentsPage`?

File: src/main/java/teammates/ui/template/CommentsForStudentsTable.java [hunk 14]
```diff
@@ -11,7 +11,7 @@
     private String giverDetails;
     private String extraClass = "";
     private List<CommentRow> rows;
-    private boolean instructorAllowedToGiveComment;
+    private boolean isInstructorAllowedToGiveComment;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/CommentsForStudentsTable.java]: `canInstructorGiveComment`?

File: src/main/java/teammates/ui/template/FeedbackResponseCommentRow.java [hunk 15]
```diff
@@ -28,15 +28,15 @@
     private Map<FeedbackParticipantType, Boolean> responseVisibilities;
 
     private String whoCanSeeComment;
-    private boolean withVisibilityIcon;
-    private boolean withNotificationIcon;
-    private boolean withLinkToCommentsPage;
+    private boolean isWithVisibilityIcon;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/FeedbackResponseCommentRow.java]: `hasVisibilityIcon`?

File: src/main/java/teammates/ui/template/FeedbackResponseCommentRow.java [hunk 16]
```diff
@@ -28,15 +28,15 @@
...
+    private boolean isWithNotificationIcon;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/FeedbackResponseCommentRow.java]: `hasNotificationIcon`?

File: src/main/java/teammates/ui/template/FeedbackResponseCommentRow.java [hunk 17]
```diff
@@ -28,15 +28,15 @@
...
+    private boolean isWithLinkToCommentsPage;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/FeedbackResponseCommentRow.java]: `hasLinkToCommentsPage`?

File: src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java [hunk 18]
```diff
@@ -22,10 +22,10 @@
     private String remindParticularStudentsPageLink;
     private String editCopyLink;
 
-    private boolean allowedToEdit;
-    private boolean allowedToDelete;
-    private boolean allowedToSubmit;
-    private boolean allowedToRemind;
+    private boolean isAllowedToEdit;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java]: `canEdit`?

File: src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java [hunk 19]
```diff
@@ -22,10 +22,10 @@
...
+    private boolean isAllowedToDelete;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java]: `canDelete`?

File: src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java [hunk 20]
```diff
@@ -22,10 +22,10 @@
...
+    private boolean isAllowedToSubmit;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java]: `canSubmit`?

File: src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java [hunk 21]
```diff
@@ -22,10 +22,10 @@
...
+    private boolean isAllowedToRemind;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java]: `canRemind`?

File: src/main/java/teammates/ui/template/InstructorStudentListStudentsTableCourse.java [hunk 22]
```diff
@@ -4,27 +4,27 @@
...
 public class InstructorStudentListStudentsTableCourse {
 
-    private boolean courseArchived;
+    private boolean isCourseArchived;
     private String courseId;
     private String courseName;
     private String googleId;
     private String instructorCourseEnrollLink;
-    private boolean instructorAllowedToModify;
+    private boolean isInstructorAllowedToModify;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/InstructorStudentListStudentsTableCourse.java]: `canBeModifiedByInstructor`?

File: src/main/java/teammates/ui/template/StudentListSectionData.java [hunk 23]
```diff
@@ -10,18 +10,18 @@
 public class StudentListSectionData {
 
     private String sectionName;
-    private boolean allowedToViewStudentInSection;
-    private boolean allowedToModifyStudent;
-    private boolean allowedToGiveCommentInSection;
+    private boolean isAllowedToViewStudentInSection;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/StudentListSectionData.java]: `canViewStudentInSection`?

File: src/main/java/teammates/ui/template/StudentListSectionData.java [hunk 24]
```diff
@@ -10,18 +10,18 @@
...
+    private boolean isAllowedToModifyStudent;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/StudentListSectionData.java]: `canModifyStudent`?

File: src/main/java/teammates/ui/template/StudentListSectionData.java [hunk 25]
```diff
@@ -10,18 +10,18 @@
...
+    private boolean isAllowedToGiveCommentInSection;
```
Comment (whipermr5) [inline src/main/java/teammates/ui/template/StudentListSectionData.java]: `canGiveCommentInSection`?

File: src/test/java/teammates/test/cases/action/FeedbackSessionStatsPageActionTest.java [hunk 26]
```diff
@@ -54,19 +54,19 @@ public void testExecuteAndPostProcess() {
                 Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId
         };
 
-        boolean doesThrowUnauthorizedAccessException = false;
+        boolean hasThrowUnauthorizedAccessException = false;
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/action/FeedbackSessionStatsPageActionTest.java]: `hasThrownUnauthorizedAccessException`

File: src/main/java/teammates/ui/pagedata/InstructorFeedbackEditPageData.java [hunk 27]
```diff
@@ -172,62 +172,62 @@ private String getDropdownMenuLabel(FeedbackQuestionAttributes question) {
     }
 
     private boolean isVisibilitySetToAnonymousToRecipientAndInstructors(FeedbackQuestionAttributes question) {
-        boolean responsesVisibleOnlyToRecipientAndInstructors = question.showResponsesTo.size() == 2
+        boolean isResponsesVisibleOnlyToRecipientAndInstructors = question.showResponsesTo.size() == 2
```
Comment (whipermr5) [inline src/main/java/teammates/ui/pagedata/InstructorFeedbackEditPageData.java]: This is now a grammatical error due to the plural `responses`. I feel the rule should include `are` and `have` as the plural counterparts to `is` and `has`. Thoughts?

Comment (xpdavid) [inline src/main/java/teammates/ui/pagedata/InstructorFeedbackEditPageData.java]: I think it should be read as this `is` `ResponsesVisibleOnlyToRecipientAndInstructors` (`true` or `false`). Use `is` for consistency. So next time when we see `is`, we would know it is a boolean variable.

Comment (whipermr5) [inline src/main/java/teammates/ui/pagedata/InstructorFeedbackEditPageData.java]: Understood; it's fine then :)

**Comment by wkurniawan07** [comment]
@xpdavid let's continue on this?

File: src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java (hunk 19 above)
Comment (xpdavid) [inline src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java]: We need to consider the getter and setter method here.
Currently, `isAllowedToDelelet` corresponds to `public boolean isAllowedToEdit()`. To make it consistent, we shall not change the expression.

Comment (whipermr5) [inline src/main/java/teammates/ui/template/InstructorFeedbackSessionActions.java]: Oh I see; it's fine then :)

**Comment by whipermr5** [comment]
LGTM; will approve after the newly-introduced merge conflict is resolved :)

**Comment by xpdavid** [comment]
Give me some time to fix the tests (if any)

**Comment by whipermr5** [comment]
@xpdavid Builds are failing due to #7450; apologies 😅

**Comment by whipermr5** [comment]
@xpdavid Another new conflict after I merged #7344 😅

**Pull request merged**